content/Serialization/Serialization.q.md

```diff
@@ -0,0 +1,1284 @@
+}
+$converted = serialize($myArray);
+$unserialized = unserialize();
+print_r($unserialized);
+?>
+```
+/// type=SS, answer=[4]
+
+Execute the program. What is the error message?
+
+ - Warning: A non-numeric value encountered on line number 6
+
+ - syntax error, unexpected `'serialize'` (T_STRING) on line number 6
+
+ - Warning: `serialize()` expects exactly `1` parameter, `0` given on line number 6
+
+ - Warning: `unserialize()` expects at least `1` parameter, `0` given on line number 7
+
+ - Warning: `unserialize()` expects parameter `1` to be string, array given on line number 7
+
+
+/// type=MS, answer=[1,4]
+
+Which statements correctly describe the error?
+
+ - On line 7, the statement `$unserialized = unserialize();` is invalid.
+
+ - There is no argument specified in the `serialize()` function call on line 6.
+
+ - On line 6, the statement `$converted = serialize($myArray);` is invalid.
+
+ - There is no argument specified in the `unserialize()` function call on line 7.
+
+ - On line 6, the argument `$myArray` in the `serialize()` function call is invalid.
+
+:::
+
+
+/// type=CR, answer=[tests/Serialization/MissingArgumentTypeForUnserializeFunctionTest.php]
+
+Correct the code so that it outputs `Array ( [1] => 1 [2] => 2 [3] => 3 [4] => 4 [5] => 5 )` in separate lines.
+
+```php
+<?php
+$number = 5;
+for ($counter = 1; $counter <= $number; $counter++) {
+    $myArray[$counter] = $counter;
+}
+$converted = serialize($myArray);
+$unserialized = unserialize();
+print_r($unserialized);
+?>
+```
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+class Worker
+{
+    private $id;
+    protected $name;
+    protected $dept;
+
+    public function __construct($id, $name, $dept)
+    {
+        $this->id = $id;
+        $this->name = $name;
+        $this->dept = $dept;
+    }
+
+    public function display()
+    {
+        echo "ID : " . $this->id . "\nName : " . $this->name . "\nDepartment : " . $this->dept;
+    }
+}
+
+$worker = new Worker(W-1002, 'Charles Han', 'Accounting');
+$serializedWorker = serialize($worker);
+print_r($serializedWorker);
+echo "\n";
+$unserializedWorker = unserialize($serializedWorker);
+print_r($unserializedWorker);
+?>
+```
+/// type=MS, answer=[1,2]
+
+Execute the program. What are the error messages?
+
+ - Warning: A non-numeric value encountered on line number 21
+
+ - Notice: Use of undefined constant `W` - assumed `'W'` on line number 21
+
+ - Warning: `serialize()` expects exactly `1` parameter, `0` given on line number 22
+
+ - Warning: `unserialize()` expects parameter `1` to be string, object given on line number 25
+
+ - syntax error, unexpected `''Accounting''` (T_CONSTANT_ENCAPSED_STRING), expecting `','` or `')'` on line number 21
+
+
+/// type=MS, answer=[2,5]
+
+Which statements correctly describe the error?
+
+ - There is no comma `,` between `Charles Han` and `Accounting` on line 21.
+
+ - On line 21, the first argument `W-1002` is not enclosed in double quotes `""`.
```

Comment:
Maybe we can use single quotes since single quotes was used in the sample code in Part 1 and the other parameters are also in single quotes.